### PR TITLE
OpenFeature Feature Flags

### PR DESCRIPTION
**Pull request by nullaus**
Feature Flag Provider with Hook. 
Server Side Feature Flag Client. 
Docker Support. 
Env updates.

### CODE REVIEW

File: packages/datum-web/package.json [hunk 1]
```diff
@@ -42,6 +42,8 @@
     "react-dom": "18.3.1"
   },
   "dependencies": {
+    "@openfeature/go-feature-flag-provider": "^0.7.3",
+    "@openfeature/server-sdk": "^1.15.1",
```
Comment (nullaus) [inline packages/datum-web/package.json:46]: I'm not sure if these were added in the correct package.json file.

**Review by lightsofapollo — APPROVED** [review]
Code LGTM - is FEATURE_FLAG_PROVIDER_URL set in vercel?

File: packages/datum-web/package.json (hunk 1 above)
Comment (angusbezzina) [reply packages/datum-web/package.json:46]: Yeah, this should be inside the package.json file for `apps/operator`

File: packages/datum-web/apps/operator/src/lib/feature-flags/feature-flags-provider.tsx
Comment (angusbezzina) [inline packages/datum-web/apps/operator/src/lib/feature-flags/feature-flags-provider.tsx:1]: We should move this file and the one beneath into `operator/src/providers/flags`

File: packages/datum-web/package.json (hunk 1 above)
Comment (nullaus) [reply packages/datum-web/package.json:46]: cool cool, will move it. i bet i ran `bun i` from the wrong directory.

File: packages/datum-web/apps/operator/src/lib/feature-flags/feature-flags-provider.tsx
Comment (nullaus) [reply packages/datum-web/apps/operator/src/lib/feature-flags/feature-flags-provider.tsx:1]: Is that OK even if it's for the server side of the web app?

**Comment by nullaus** [comment]
> Code LGTM - is FEATURE_FLAG_PROVIDER_URL set in vercel?

Yes indeed. I went ahead and did all of the Vercel and Koyeb configuration yesterday as part of this work.

**Review by angusbezzina — APPROVED** [review]
Looks good to me, just one more nit (if you want to) - you can probably remove provider from the file name for feature-flag-providers since it's sitting in the providers folder it should be clear that it's a provider

**Comment by nullaus** [comment]
> Looks good to me, just one more nit (if you want to) - you can probably remove provider from the file name for feature-flag-providers since it's sitting in the providers folder it should be clear that it's a provider

Angus and I chatted and we're going to leave the filename as is to avoid having to type `.tsx` to ensure it loads the correct file and provider.

**Pull request merged**